sys/net/routing/ng_rpl/ng_rpl_control_messages.c

```diff
@@ -595,25 +616,34 @@ void ng_rpl_recv_DAO(ng_rpl_dao_t *dao, ipv6_addr_t *src, uint16_t len)
 {
     ng_rpl_instance_t *inst = NULL;
     ng_rpl_dodag_t *dodag = NULL;
+    ng_rpl_opt_t *opts = (ng_rpl_opt_t *) (dao + 1);
+
     if ((inst = ng_rpl_instance_get(dao->instance_id)) == NULL) {
         DEBUG("RPL: DAO with unknown instance id (%d) received\n", dao->instance_id);
         return;
     }
 
+    len -= (sizeof(ng_rpl_dao_t) + sizeof(icmpv6_hdr_t));
```

Comment:
Actually, none of the receive functions check the length of the packets before processing them. This was an oversight when reviewing #3050 and needs to be fixed, but it can be done in a separate PR if you prefer to keep this PR more to the point.